github-crawler-autoconfigure/src/test/java/com/societegenerale/githubcrawler/BitbucketCrawlerIT.java

```diff
@@ -0,0 +1,335 @@
+package com.societegenerale.githubcrawler;
+
+import com.societegenerale.githubcrawler.config.GitHubCrawlerAutoConfiguration;
+import com.societegenerale.githubcrawler.config.TestConfig;
+import com.societegenerale.githubcrawler.mocks.BitbucketMock;
+import com.societegenerale.githubcrawler.model.Repository;
+import org.assertj.core.api.SoftAssertions;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.io.IOException;
+import java.util.*;
+
+import static com.jayway.awaitility.Awaitility.await;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toSet;
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = {TestConfig.class, GitHubCrawlerAutoConfiguration.class})
+@ActiveProfiles(profiles = {"bitBucketTest"})
+@DirtiesContext(classMode = ClassMode.BEFORE_CLASS)
+public class BitbucketCrawlerIT {
+
+  private final int MAX_TIMEOUT_FOR_CRAWLER = 50;
+
+  @Autowired
+  private GitHubCrawler crawler;
+
+  private static BitbucketMock bitbucketMockServer;
+
+  @Autowired
+  private BitbucketMock tmpBitbucketMockServer;
+
+  @Autowired
+  private TestConfig.InMemoryGitHubCrawlerOutput output;
+
+  private static boolean hasBitbucketMockServerStarted = false;
+
+  private int nbRepositoriesInOrga = 4;
+
+  @BeforeEach
+  public void mockSetUp() {
+
+    //we need a bitbucketMockServer to be static, so that we can stop it once at the end in the @AfterClass method
+    //but at the same time, we need a version of it that is configured by Spring, so can't be static
+
+    //therefore we have small hack below, where we take the reference of the BitBucketMock configured and injected by Spring,
+    //and assign it to the static variable the first time, when the static reference is null
```

Comment:
you took this from GitHubCrawlerIT, but I am not sure it's really needed, even in GitHubCrawlerIT actually. I will try to remove it in both afterward

Reply:
see https://github.com/societe-generale/github-crawler/commit/a883e6c9da3c30e5a6073f1bf7532c783347d8b3